lib/auth/trust/trustv1/service.go

```diff
@@ -27,13 +28,21 @@ import (
 	"github.com/gravitational/teleport/lib/services"
 )
 
+type authServer interface {
+	// GenerateHostCert uses the private key of the CA to sign the public key of
+	// the host (along with meta data like host ID, node name, roles, and ttl)
+	// to generate a host certificate.
+	GenerateHostCert(ctx context.Context, hostPublicKey []byte, hostID, nodeName string, principals []string, clusterName string, role types.SystemRole, ttl time.Duration) ([]byte, error)
```

Comment:
nit:  8 params in function call is to much for me but leaving it up to you.

Reply:
I agree - but this is actually the existing signature. My plan is to probably change this at a later date when the deprecated methods are removed in 16.0.0